testing/xfpga/test_sysfs_c.cpp

```diff
@@ -554,3 +475,161 @@ TEST(sysfs_c, cstr_dup) {
 
 INSTANTIATE_TEST_CASE_P(sysfs_c, sysfs_c_p,
                         ::testing::ValuesIn(test_platform::keys(true)));
+
+class sysfs_c_hw_p : public sysfs_c_p {
+ protected:
+  sysfs_c_hw_p() {}
+};
+
+/**
+ * @test    make_sysfs_group
+ * @details
+ */
+TEST_P(sysfs_c_hw_p, make_sysfs) {
+  const std::string invalid_path =
+      "/sys/class/fpga/intel-fpga-dev.0/intel-fpga-fme";
+  _fpga_token *tok = static_cast<_fpga_token *>(tokens_[0]);
+  fpga_object obj;
+  auto res = make_sysfs_group(tok->sysfspath, "errors", &obj, 0, handle_);
+  EXPECT_EQ(res, FPGA_OK);
+  EXPECT_EQ(xfpga_fpgaDestroyObject(&obj), FPGA_OK);
+
+  res = make_sysfs_group(tok->sysfspath, "errors", &obj, FPGA_OBJECT_GLOB,
+                         handle_);
+  EXPECT_EQ(res, FPGA_OK);
```

Comment:
Will obj be leaked when line 505 executes?

Reply:
Probably. Will fix.